amazon/ion/ioncmodule.c

```diff
@@ -775,7 +779,9 @@ static PyObject* ionc_write(PyObject *self, PyObject *args, PyObject *kwds) {
     PyObject *obj, *binary, *sequence_as_stream, *tuple_as_sexp;
     ION_STREAM  *ion_stream = NULL;
     BYTE* buf = NULL;
+    hWRITER writer = NULL;
```

Comment:
This was the more concerning warning we had. With the use of `IONCHECK`'s gotos on error, there's a lot of potential to jump around initializations causing potential UB. We should try to keep variable declarations at the head of the function when possible, to ensure initialization occurs for all branches.